### PR TITLE
Add time warp patch hard fork.

### PR DESCRIPTION
**Pull request by toxeus**

### CODE REVIEW

File: src/chain/chain_state.cpp [hunk 1]
```diff
@@ -191,6 +194,9 @@ chain_state::activations chain_state::activation(const data& values,
         result.minimum_block_version = settings.first_version;
     }
 
+    if (time_warp)
+        result.forks |= (rule_fork::time_warp_patch & forks);
+
```
Comment (evoskuil) [inline src/chain/chain_state.cpp]: Remove above (dead code).

File: src/chain/chain_state.cpp [hunk 2]
```diff
@@ -481,8 +487,10 @@ chain_state::data chain_state::to_pool(const chain_state& top,
     // Alias configured forks.
     const auto forks = top.forks_;
 
-    // Retargeting is only activated via configuration.
+    // Retargeting and time_warp_patch are only activated via configuration.
     const auto retarget = script::is_enabled(forks, rule_fork::retarget);
+    const auto time_warp = script::is_enabled(forks, rule_fork::time_warp_patch
+        );
```
Comment (evoskuil) [inline src/chain/chain_state.cpp]: Move inside `if (retarget &&...` condition scope, or even just eliminate intermediate constant.

File: src/chain/chain_state.cpp (hunk 1 above)
Comment (toxeus) [reply src/chain/chain_state.cpp]: Done.

File: src/chain/chain_state.cpp (hunk 2 above)
Comment (toxeus) [reply src/chain/chain_state.cpp]: Done.

File: src/chain/chain_state.cpp [hunk 3]
```diff
@@ -126,6 +127,10 @@ chain_state::activations chain_state::activation(const data& values,
     // bip90 is activated based on configuration alone (hard fork).
     result.forks |= (rule_fork::bip90_rule & forks);
 
+    // time_warp_patch is activated based on configuration alone (hard fork).
+    result.forks |= (rule_fork::time_warp_patch & forks);
+
+
```
Comment (evoskuil) [inline src/chain/chain_state.cpp]: whitespace (extra line)

**Pull request merged**